service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -44,18 +45,18 @@
      * @param connectionString The IoTHub connection string
      */
     @Builder(builderMethodName = "buildFromConnectionString", builderClassName = "FromConnectionStringBuilder")
-    public DigitalTwinAsyncClient(@NonNull String connectionString) {
-        ServiceConnectionString serviceConnectionString = ServiceConnectionStringParser.parseConnectionString(connectionString);
-        SasTokenProvider sasTokenProvider = serviceConnectionString.createSasTokenProvider();
-        String httpsEndpoint = serviceConnectionString.getHttpsEndpoint();
+    public DigitalTwinAsyncClient(@NonNull String connectionString) throws IOException {
+        IotHubConnectionString iotHubConnectionString = IotHubConnectionStringBuilder.createConnectionString(connectionString);
+        IotHubServiceSasToken iotHubServiceSasToken = new IotHubServiceSasToken(iotHubConnectionString);
+        String httpsEndpoint = "https://" + iotHubConnectionString.getHostName();
```

Comment:
is there not a way in Java to create a well formed URI? String concatenation makes me feel uneasy.

Reply:
I don't see it done other ways. I will try to google some more. But would be great to know from others who code in java.